### ISSUE TITLE
Negative loss

### ISSUE THREAD

**Issue by OuYangg** [Correr-Zhou/SPML-AckTheUnknown#1]
Thank you very much for open sourcing this great work!

I have included negative examples with all labels set to 0 during my training process. As a result, when a batch consists only of negative samples, the calculated loss becomes negative. I would like to ask if you also incorporate negative examples with all labels set to 0 in your training process, and how do you resolve this issue?

```
LOG_EPSILON = 1e-5

def neg_log(x):
    return - torch.log(x + LOG_EPSILON)

class EntropyMaximization(nn.Module):
    ## 负样本全为0，此时loss会变负数
    def __init__(self, alpha: float=0.25):
        super(EntropyMaximization, self).__init__()
        self.alpha = alpha
    
    def forward(self, logits: torch.Tensor, targets: torch.Tensor):
        logits = torch.sigmoid(logits)
        
        loss_denom_mtx = (logits.shape[0] * logits.shape[1]) * torch.ones_like(logits)
        loss_mtx = torch.zeros_like(logits)
        loss_mtx[targets==1] = neg_log(logits[targets==1])
        loss_mtx[targets==0] = -self.alpha * (
            logits[targets==0] * neg_log(logits[targets==0]) + 
            (1-logits[targets==0]) * neg_log(1-logits[targets==0])
        )
        loss = (loss_mtx / loss_denom_mtx).sum()
        return loss
```
![image](https://github.com/user-attachments/assets/708e4fe7-394b-40ff-8060-ef6fd64f1d7d)

**Comment by Correr-Zhou** [comment]
Hi, thank you for asking. : )
Yes, we set all unannotated labels to 0 (including potential negative labels).
It is normal for the losses of some batches to be below 0, which would not influence the training dynamics of EM loss.
Just imagine that you add a large constant to the losses, and it would turn the losses to be above 0 but has no effect to the loss gradients.

**Comment by Correr-Zhou** [comment]
I will close this issue. Feel free to re-open it if you have another question. : )
